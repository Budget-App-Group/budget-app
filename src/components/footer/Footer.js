import React from "react";
// import { Link } from "react-router-dom";
import "./footer.scss";

export default function Footer() {
  return (
    <div className="footer-main">
      <div className="footer-copy"> Â© Whitney, Nick, Colby</div>
      <div className="footer-links">
        {/* <Link to="/contactus">Contact Us</Link> */}
      </div>
    </div>
  );
}
