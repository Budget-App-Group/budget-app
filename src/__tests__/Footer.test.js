import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Footer from "../components/footer/Footer";
import { MemoryRouter } from "react-router";

//contain tests
it("footer does show copyright when mounted", () => {
  const { container } = render(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  );

  expect(container.textContent).toContain("Â©");
});

it("footer does show names when mounted", () => {
  const { container } = render(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  );

  expect(container.textContent).toContain("Whitney, Nick, Colby");
});

it("footer does contain Link when mounted", () => {
  const { container } = render(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  );

  expect(container.textContent).toContain("Contact Us");
});

//snapshot tests
describe("Snapshot test of footer", () => {
  const component = renderer.create(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  );
  let tree = component.toJSON();

  test("component Footer matches the snapshot when rendered", () => {
    expect(tree).toMatchSnapshot();
  });

  test("component Footer renders truthy each time rendered", () => {
    expect(tree).toBeTruthy();
  });
});
