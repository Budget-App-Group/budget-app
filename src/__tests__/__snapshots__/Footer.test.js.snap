// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of footer component Footer matches the snapshot when rendered 1`] = `
<div
  className="footer-main"
>
  <div
    className="footer-copy"
  >
     Â© Whitney, Nick, Colby
  </div>
  <div
    className="footer-links"
  >
    <a
      href="/contactus"
      onClick={[Function]}
    >
      Contact Us
    </a>
  </div>
</div>
`;
